cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/aws-mc-common/default-service/AwsDefaults.service.tsx

```diff
@@ -25,4 +25,47 @@ export class AwsDefaults {
         }
         return nodeTypeList[Math.round(nodeTypeList.length / 2)];
     };
+
+    static defaulAvailabilityZoneNameStrategy = (azList: { [key: string]: string }[], nodeProfile: string) => {
+        const defaultAZNameList: string[] = [];
+        switch (nodeProfile) {
+            case AWS_NODE_PROFILE_NAMES.SINGLE_NODE: {
+                return [azList[0].id];
+            }
+            default: {
+                azList.slice(0, 3).forEach((az) => {
+                    defaultAZNameList.push(az.id);
+                });
+                return defaultAZNameList;
+            }
+        }
+    };
+
+    static defaulAvailabilityZoneNodeTypeStrategy = (azList: { [key: string]: string }[], nodeProfile: string) => {
+        const defaultNodeTypes = getDefaultNodeTypes(nodeProfile);
+        for (const az of azList) {
+            for (let i = 0; i < defaultNodeTypes.length; i++) {
+                if (az.workerNodeType === defaultNodeTypes[i]) {
+                    return az;
+                }
+            }
+        }
+
+        const defaultNodeTypeByAZ: { [key: string]: any } = {
+            workerNodeType: defaultNodeTypes[Math.round(defaultNodeTypes.length / 2)],
+            publicSubnetID: '',
+            privateSubnetID: '',
+        };
+
+        return defaultNodeTypeByAZ;
+    };
+
+    static async createAZNodeType(defaultAZName: string) {
+        const azNodeTypes: { [key: string]: string }[] = [];
+        const nodeTypes = await AwsService.getAwsNodeTypes(defaultAZName);
+        nodeTypes.forEach((nodeType) => {
+            azNodeTypes.push({ name: defaultAZName, workerNodeType: nodeType, publicSubnetID: '', privateSubnetID: '' });
```

Comment:
Yes sir. We are getting the node instance data and transforming them to a strucutre.
Created an interface for Availability Zone.
```
export interface AvailabilityZoneInstance {
    name: string;
    workerNodeType: string;
    publicSubnetID: string;
    privateSubnetID: string;
}
```
Use `nodeTypes.map` instead of `forEach`